pkg/auth/authz_handler.go

```diff
@@ -46,20 +47,124 @@ func (a *AuthzHandler) CreateAuthorizer() func(*http.Request) error {
 	return a.Authorizer
 }
 
-func (a *AuthzHandler) isObjectOwnedByUser(id string, obj interface{}, username string) (bool, error) {
+func (a *AuthzHandler) isTenancyEnabled() bool {
+	//MGMT-9443 TODO: query here feature flag
+	return true
+}
+
+func handleOwnershipQueryError(err error) (bool, error) {
+	if err != nil {
+		//if user is not the owner of the object return false
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return false, nil
+		}
+		//in case of a real db error, indicate it to the caller
+		return false, err
+	}
+	return true, nil
+}
+
+func (a *AuthzHandler) isObjectOwnedByUser(id string, obj interface{}, payload *ocm.AuthPayload) (bool, error) {
+	if a.db != nil {
+		err := a.db.First(obj, "id = ? and user_name = ?", id, payload.Username).Error
+		return handleOwnershipQueryError(err)
+	}
+
+	return true, nil
+}
+
+func (a *AuthzHandler) isObjectOwnedByOrg(id string, obj interface{}, payload *ocm.AuthPayload) (bool, error) {
 	if a.db != nil {
-		err := a.db.First(obj, "id = ? and user_name = ?", id, username).Error
+		err := a.db.First(obj, "id = ? and org_id = ?", id, payload.Organization).Error
```

Comment:
maybe worth caching this as well. to avoid redundant calls to db.

Reply:
on a separate PR